hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/batch/CompositeBatchPersister.java

```
@@ -71,10 +71,11 @@ public void persist(Collection<? extends Object> items) {
         }
 
         Object item = items.iterator().next();
-        BatchPersister batchPersister = batchInserters.get(item.getClass());
+        Class<?> itemClass = item != null ? item.getClass() :null;
```

Why not just exit early on null? Just like we do when items is null or empty?

---

I don't think we should silently ignore using a `return` statement like null or empty. Having a null in the item list is an unsupported scenario that indicates a coding error. Imagine if the it was `List.of(null, new Contract(), new Contract())` then the last two valid contracts would be discarded.

Unless you're saying we should have a separate if item is null then throw exception? I didn't want to duplicate the exception throwing code so went with the less code approach.

---

Actually you're right. Safer this way too. Carry on